src/App.js

```diff
@@ -1,10 +1,50 @@
+import {
+  createBrowserRouter,
+  createRoutesFromElements,
+  Route,
+  RouterProvider,
+  Outlet, NavLink,
+} from 'react-router-dom';
 import './App.css';
+import Books from './components/Books';
+import Categories from './components/Categories';
+
+function Layout() {
+  return (
+    <>
+      <nav>
+        <h1 className="heading">Bookstore</h1>
+        <ul>
+          <li>
+            <NavLink to="/">
+              Books
+            </NavLink>
+          </li>
+          <li>
+            <NavLink to="categories">
+              Categories
+            </NavLink>
+          </li>
+        </ul>
+      </nav>
+      <Outlet />
+    </>
+  );
+}
```

Comment:
- [optional] I think you can implement the layout function in a separate file, this will make your app.js cleaner. App.js should serves as container for all other components. ℹ️